### PR TITLE
feat: support additional newline characters

### LINKED ISSUE #11: Handle other types of newline
Only `\n` is currently supported as termination of nodes.

All newline characters from the spec should be supported: https://github.com/kdl-org/kdl/blob/main/SPEC.md#newline

### PR DESCRIPTION
**Pull request by focusaurus**
fixes #11

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -178,8 +178,13 @@ func scanNode(cx *parseContext, sc *pkg.Scanner, name string) (Node, error) {
...
 		case pkg.WS:
-			// FIXME: handle other types of newline control characters (see the spec)
-			if strings.Contains(lit, "\n") || strings.Contains(lit, "\r") {
+			if strings.Contains(lit, "\n") || // newline
+				strings.Contains(lit, "\r") || // carriage return
+				strings.Contains(lit, "\r\n") || // carriage return newline
+				strings.Contains(lit, "\f") || // form feed
+				strings.Contains(lit, "\u0085") || // next line
+				strings.Contains(lit, "\u2028") || // line separator
+				strings.Contains(lit, "\u2029") { // paragraph separator
```
Comment (lunjon) [inline parser.go:187]: I think this must be handled as well in the `scanString` method below.
If you put all these possible "newline" characters into an array or something (global variable, perhaps)
it should be fairly straightforward.

**Comment by lunjon** [comment]
This is so kind of you. Thank you very much!

File: parser.go (hunk 1 above)
Comment (focusaurus) [reply parser.go]: OK I updated `scanString` as well but the unit test I added is failing for line separator and paragraph separator in quoted string literal values with an EOF error. Not sure if that's a parsing bug that needs to be fixed or a bug in this change. Could you have a look?

Comment (lunjon) [reply parser.go]: Awesome! I can take a look .

Comment (lunjon) [reply parser.go]: I found the issue. Adding the following case to the switch in scanString should do the trick:
```go
		case pkg.RAWSTR_OPEN, pkg.RAWSTR_HASH_OPEN:
			buf.WriteString(lit[:len(lit)-1])
			done = true
```

**Comment by focusaurus** [comment]
Awesome! I incorporated your fix so this PR I think should be ready now.

**Review by lunjon — APPROVED** [review]
# :star:

**Pull request merged**